### PR TITLE
Scenegraph benchmark

### PR DESCRIPTION
**Pull request by lcampagn**
WIP: This PR adds a benchmark example for comparing scenegraph performance to similar optimal implementations. 

The intent is to add more benchmark examples and optimizations until we get comparable performance (or decide that we will never get comparable performance).

### CODE REVIEW

File: vispy/visuals/shaders/function.py [hunk 1]
```diff
@@ -430,7 +431,7 @@ def __setitem__(self, key, val):
                     self.template_vars.add(var.lstrip('$'))
         
         self.changed(code_changed=True, value_changed=True)
-        if logger.level >= 10:
+        if logger.level == logging.DEBUG:
```
Comment (Eric89GXL) [inline vispy/visuals/shaders/function.py]: I think this could be `<= logging.DEBUG` in case we ever make/use a custom logging level that is below DEBUG

**Comment by Eric89GXL** [comment]
@lcampagn do you need any other info to progress on this PR, or is it ready (enough) for merge? It seemed like we were close to merge at some point, and there are some useful changes in here. WDYT?

**Comment by lcampagn** [comment]
It was ready long ago.. we'll see how the tests work out now :)

**Comment by Eric89GXL** [comment]
I exepect AppVeyor to fail, if you want to see full success go ahead and rebase, otherwise we can just ignore AppVeyor for now.

**Comment by Eric89GXL** [comment]
You'll probably need a rebase to get the tests to run properly

**Comment by lcampagn** [comment]
Don't the tests run on the merged code? If yes, then the tests should come out identical. If no, then there's obviously a big problem :)

**Comment by Eric89GXL** [comment]
Unfortunately I don't think they do...

**Comment by lcampagn** [comment]
If that were the case, then we would expect many PRs to show passing tests while open, but then break tests in master after merging.

**Comment by Eric89GXL** [comment]
Looks like you're right, everybody is happy

**Comment by Eric89GXL** [comment]
Thanks, one more PR cleared :)

**Pull request merged**